Sources/Basic/StringConversions.swift

```diff
@@ -0,0 +1,54 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright 2015 - 2016 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+/// Whitelisted characters which do not need any shell escaping.
+private let whitelist: Set<UInt8> = {
+    Set("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ-_/:@#%+=.,".utf8.map{$0})
+}()
+
+public extension String {
+
+    /// Creates a shell escaped string. If the string does not need escaping, returns the original string.
+    /// Otherwise escapes using single quotes. For eg: hello -> hello, hello$world -> 'hello$world', input A -> 'input A'
+    ///
+    /// - Returns: Shell escaped string.
+    public func shellEscaped() -> String {
+
+        // If all the characters in the string are in whitelist then no need to escape.
+        guard let pos = utf8.index(where: { !whitelist.contains($0) }) else {
+            return self
+        }
+
+        // If there are no single quotes then we can just wrap the string around single quotes.
+        guard let singleQuotePos = utf8[pos..<utf8.endIndex].index(of: 39) else {
+            return "'" + self + "'"
+        }
+
+        // Otherwise iterate and escape all the single quotes.
+        var newString = "'" + String(utf8[utf8.startIndex..<singleQuotePos])
+
+        for char in utf8[singleQuotePos..<utf8.endIndex] {
+            if char == 39 {
+                newString += "\\'"
```

Comment:
This doesn't work for all cases, at least not in Bash.  Backslash doesn't work that way in Bash when single quotes are involved.

Consider `'\''`.  One would expect that to result in just `'` but instead it's a `\` followed by an unterminated single quote.

What you actually need to do is to end the single quote sequence, then have an escaped single quote, then start a new single quote sequence.

So a string like `Apple's` becomes `'Apple'\''s'` when escaped.  It's kind of weird and unfortunate, but single quote is like this.  Double quote is much more normal, but of course won't quote nearly as many characters.